clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java

```diff
@@ -140,8 +140,8 @@ public void ensureValid() {
      *
      * @return The base timestamp
      */
-    public long baseTimestamp() {
-        return buffer.getLong(BASE_TIMESTAMP_OFFSET);
+    public long firstTimestamp() {
```

Comment:
Ack. I will also update the javadoc in this class to explain the difference in usage between `baseOffset` and `firstTimestamp`